### DISCUSSION TITLE
Polars Backend

### DISCUSSION THREAD

**Discussion by johalnes** [casact/chainladder-python#520] [General]
Some discussions around Polars backend was in #482 and @jbogaardt [mentioned](https://github.com/casact/chainladder-python/issues/474#issuecomment-1857830127) that some experimentation was done, but looks stale at the moment. 

Having looked at both Polars plugin in Rust as well as other projects, this seems even more tempting now than before. Especially with packages like the ones below showing how elegant much code gets:
* [Polars-ols](https://github.com/azmyrajab/polars_ols?tab=readme-ov-file)
* [Polars-ds](https://github.com/abstractqqq/polars_ds_extension) 
* [functime](https://github.com/functime-org/functime/tree/main)

Has anyone thought or experimented any more with this topic? 🙂 And maybe the biggest question - should this be new major release within this package, with some changes to the API, or should this actually be a new project like chainladder-pl?

**Comment by jbogaardt** [comment]
Hi @johalnes, I think its a great idea and I'm for whichever approach (major release, new package) nets broader community involvement in advancing reserving capabilities in python. It is stale because I've personally not had the time to commit to chainladder development as I once used to. In either case, happy to collaborate where I can.

**Comment by johalnes** [comment]
I feel the lack of time myself, so for the development it’ll probably best to have it within this package so more people may contribute. So impressed by the work you and the other have done!

if there should be a major update, is there anything else you would want to change? Any design decisions you have some regrets about?

I’ll think dates is better handled in Polars, together with only one backend, I would think the project will be somewhat simplified 🙂
